autoPyTorch/pipeline/components/training/data_loader/image_data_loader.py

```diff
@@ -0,0 +1,61 @@
+from typing import Any, Dict
+
+import torchvision
+
+from autoPyTorch.pipeline.components.training.data_loader.base_data_loader import BaseDataLoaderComponent
+
+
+class ImageDataLoader(BaseDataLoaderComponent):
+    """This class is an interface to the PyTorch Dataloader.
+
+    Particularly, this data loader builds transformations for
+    image data.
+
+    """
+
+    def build_transform(self, X: Dict[str, Any], train: bool = True) -> torchvision.transforms.Compose:
+        """
+        Method to build a transformation that can pre-process input data
+
+        Args:
+            X (X: Dict[str, Any]): Dependencies needed by current component to perform fit
+            train (bool): whether transformation to be built are for training of test mode
+
+        Returns:
+            A composition of transformations
+        """
+
+        transformations = []
+
+        if train:
+            transformations.append(X['image_augmenter'])
+        # In the case of image data, the options currently available
+        # for preprocessors are:
+        #   + normalise
+        # These can apply for both train/val/test, so no
+        # distinction is performed
+
+        if not X['is_small_preprocess']:
```

Comment:
Could you add documentation what this means?

Reply:
yeah sure